src/cluster.c

```diff
@@ -930,10 +939,11 @@ ShardGroup *ShardGroupParse(RedisModuleCtx *ctx, RedisModuleString **argv, int a
     sg->slot_ranges_num = num_slots;
     sg->slot_ranges = RedisModule_Calloc(num_slots, sizeof(ShardGroupSlotRange));
     for (int i = 0; i < num_slots; i++) {
-        long long start_slot, end_slot, type;
+        long long start_slot, end_slot, type, key;
         if (RedisModule_StringToLongLong(argv[argidx++], &start_slot) != REDISMODULE_OK ||
             RedisModule_StringToLongLong(argv[argidx++], &end_slot) != REDISMODULE_OK ||
             RedisModule_StringToLongLong(argv[argidx++], &type) != REDISMODULE_OK ||
+            RedisModule_StringToLongLong(argv[argidx++], &key) != REDISMODULE_OK ||
```

Comment:
we talked about validating that key is >= 0 so it an be enforced as unsigned cleanly?

Reply:
we can add this validation, sure.